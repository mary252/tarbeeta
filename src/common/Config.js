// export const baseURL = "https://api.tarbeeta.com/v1/";
// export const v2_url = "https://api.tarbeeta.com/v2/";
export const v2_url = "https://staging.tarbeeta.com/v2/";
export const baseURL = "https://staging.api.tarbeeta.com/v1/";
export const socketUrl = "https://socket.tarbeeta.com:8878";
// export const baseURL = "http://192.168.1.15:3000/v1/";
// export const baseURL = "http://197.167.99.223:3126/v1/";
 //export const baseURL = "http://msi-laptop:3000/v1/";
export const assetUrl = "https://static.tarbeeta.com/images/";
export const baseUrlStatic = "https://www.tarbeeta.com";

//export const google_capcha="6Lf34qkUAAAAAKBh0Qs_9OZrBeZhikwqF6L7G62p"
// export const google_capcha="6LfE5akUAAAAAHC3KCKqJh1NGup58xJwQbjZxYgg"
export const google_capcha = "6LfXz6oUAAAAABbSRcTAPsK971UiPQZ6CpSSPWr4";
export const EN = 1;
export const AR = 2;
export const EG = 1;
export const DEFAULT_LOCALE = "ar";

// Facebook
export const FACEBOOK_APP_ID = "520130485093535";
export const FACEBOOK_PAGE_ID = "1996928467010956";
export const FACEBOOK_CHAT_POPUP_THEME_COLOR = "#5a9ae5";

export const PRODUCT_SLIDER_LIMIT = 18;

export const ORDER_STATUS_FAILED = 1;
export const ORDER_STATUS_PLACED = 2;
export const ORDER_STATUS_PAID = 3;
export const ORDER_STATUS_PENDING_PAYMENT = 4;
export const ORDER_STATUS_PENDING_SHIPPING = 5;
export const ORDER_STATUS_ON_THE_WAY = 6;
export const ORDER_STATUS_DELIVERED = 7;
export const ORDER_STATUS_CANCELLED = 8;
export const ORDER_STATUS_PENDING = 9;

export const APPROVED_PRODUCT = 4;

export const CASH_ON_DELEVERY = 4;
export const PAY_WALLET = 3;
export const MASTER_CARD = 1;
export const KIOSK = 2;

export const SORT_MOST_RECENT = 1;
export const SORT_OLD_TO_NEW = 2;
export const SORT_HIGHEST_PRICE = 3;
export const SORT_LOWEST_PRICE = 4;
export const SORT_RATING = 5;
export const CARD_PAYMENT_IFRAME = `https://accept.paymobsolutions.com/api/acceptance/iframes/`;
export const EN_FRAME_ID = 7340;
export const AR_FRAME_ID = 7459;
export const pending_product = 1;
