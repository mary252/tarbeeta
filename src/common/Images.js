const Images = {
  aman: require("../assets/images/bitmapAman.png"),
  masary: require("../assets/images/bitmap.png"),
  creditCard: require("../assets/images/cards.png"),
  imagePlaceHolder: require("../assets/images/image-grey-copy-2.png"),
  imagePlaceHolder2: require("../assets/images/image-grey-copy-2@3x.png"),

  slider8Ar: require(`../assets/images/sliders/desktop/ar/8.png`),
  slider8ArMobile: require(`../assets/images/sliders/mobile/ar/8.png`),

  slider8En: require(`../assets/images/sliders/desktop/en/8.png`),
  slider8EnMobile: require(`../assets/images/sliders/mobile/en/8.png`),

  slider9Ar: require(`../assets/images/sliders/desktop/ar/9.png`),
  slider9ArMobile: require(`../assets/images/sliders/mobile/ar/9.png`),

  slider9En: require(`../assets/images/sliders/desktop/en/9.png`),
  slider9EnMobile: require(`../assets/images/sliders/mobile/en/9.png`),

  superMarketAr: require(`../assets/images/sliders/desktop/ar/supermarket-banner-02-ar.jpg`),
  superMarketEn: require(`../assets/images/sliders/desktop/en/supermarket-banner-05-en.jpg`),

  sliderSMMobileAr: require(`../assets/images/sliders/mobile/ar/supermarket-banner-04-ar.jpg`),
  sliderSMMobileEn: require(`../assets/images/sliders/mobile/en/supermarket-banner-06-en.jpg`),

  slider10Ar: require(`../assets/images/sliders/desktop/ar/10.gif`),
  slider10ArMobile: require(`../assets/images/sliders/mobile/ar/10.gif`),

  slider10En: require(`../assets/images/sliders/desktop/en/10.gif`),
  slider10EnMobile: require(`../assets/images/sliders/mobile/en/10.gif`),

  redHeart: require("../assets/images/read-heart.svg"),
  fillHeart: require("../assets/images/heart_filled.svg"),

  notFound404: require("../assets/images/404.svg"),
  noOrders: require("../assets/images/no-orders.svg"),
  emptyCart: require("../assets/images/empty-cart.svg"),
  emptyNotifications: require("../assets/images/empty-notifications.svg"),
  emptySearch: require("../assets/images/empty-search.svg"),
  emptyFavourites: require("../assets/images/no-favourites.svg")
};

export { Images };
