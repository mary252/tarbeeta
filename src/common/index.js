export * from "./Images";
export * from "./Config";
