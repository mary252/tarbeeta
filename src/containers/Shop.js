import React, { Component } from "react";
import { connect } from "react-redux";
import {
  Page,
  Filters,
  Cover,
  FiltersMobile,
  FollowersPopup,
  ActiveModal,
  Loader,
  EmptyPage
} from "../components";
import ProductsGrid from "./Products/Grid";
import * as actions from "../actions";
import {
  getInfo,
  fetchShopFilter,
  isMyShop,
  getShopProduct,
  followShop,
  unFollowShop,
  getFollowes
} from "../services";

export class Shop extends Component {
  state = {
    id: null,
    products: [],
    filters: [],
    loading: false,
    Color: [],
    Size: [],
    Material: [],
    Subcategories: [],
    filtersBar: [],
    sort: null,
    min_price: null,
    max_price: null,
    is_mine: false,
    no_shop: false,
    filter_mobile: false,
    n_followers: 0,
    is_followed: false,
    followers: null,
    page: 1,
    is_busy: false,
    showMore: false,
    will_follow: true,
    show_loader: false,
    is_following_load: false,
    count: 0
  };

  FollowersPopup = null;
  openPopUp = () => this.FollowersPopup.toggle();

  openFilterMobile = () => {
    this.setState({
      filter_mobile: true
    });
  };

  closeFilterMobile = () => {
    this.setState({
      filter_mobile: false
    });
  };

  getFollowers = async () => {
    try {
      let res = await getFollowes(this.state.id);
      this.setState({
        followers: res.data
      });
    } catch (e) {}
  };

  componentDidMount = async () => {
    this.setState({ loading: true });
    var langId = this.props.lang_id;
    window.scroll({ top: 0, left: 0, behavior: "smooth" });

    let shop_id = this.props.match.params.id;
    await this.getShop(shop_id, langId);

    // await this.checK(shop_id);

    await this.getFilters(this.state.id, langId);
    await this.getFollowers();
    this.setState({ loading: false });
  };

  checK = async shop_id => {
    try {
      let res = await isMyShop(shop_id);
      this.setState({ isMine: true });
    } catch (e) {}
  };

  getProducts = async willMerge => {
    try {
      // is_busy : this for Dots loader in infinite
      // show_loader : for showing the circle loader

      // this.setState({ is_busy: true, show_loader: true });

      this.state.show_loader
        ? this.setState({ is_busy: false })
        : this.setState({ is_busy: true });

      // this.setState({ is_busy: true });

      let { lang_id } = this.props,
        {
          Color,
          Size,
          Material,
          sort,
          max_price,
          min_price,
          Subcategories,
          page
        } = this.state,
        filter = {
          shop_id: this.state.id,
          Colour: Color,
          Size,
          Material,
          lang_id,
          sort,
          max_price,
          min_price,
          subcategories: Subcategories,
          page
        };

      let res = await getShopProduct(this.state.id, filter);
      this.setState(state => ({
        products: willMerge ? [...state.products, ...res.data] : res.data,
        is_busy: false, // this for Dots loader in infinite
        showMore: res.showMore,
        loading: false, // this for showing the circle loader,
        count: res.count
      }));
    } catch (e) {}
  };

  getShop = async (shop_username, langId) => {
    try {
      let res = await getInfo(shop_username, langId),
        { products: { data, showMore, count } = {} } = res.data;

      this.setState({
        ...res.data,
        is_followed: Boolean(res.data.is_followed),
        count,
        products: data,
        showMore
      });
    } catch (e) {
      this.setState({ no_shop: true });
    }
  };

  getFilters = async (shop_id, langId) => {
    try {
      let res = await fetchShopFilter(
        {
          lang_id: langId
        },
        shop_id
      );

      const { filters, max_price, min_price } = res.data;

      this.setState({ filters, max_price, min_price });
    } catch (e) {}
  };

  sort = e => this.setState({ sort: e.target.value }, this.getProducts);

  saveToState = e => this.setState({ [e.target.name]: e.target.value });

  extract_filters = () => {};

  select_filter = (key, isSelected, val, obj) =>
    this.setState(state => {
      return {
        [key]: isSelected
          ? state[key].filter(a => a !== val)
          : [...state[key], val],
        filtersBar: isSelected
          ? state.filtersBar.filter(a => !(a.name == key && a.id == val))
          : [...state.filtersBar, obj],
        loading: true // this for showing the circle loader
      };
    }, this.getProducts);

  goToDetails = id =>
    this.props.history.push(`/${this.props.lang}/products/${id}`);
  navigate = () => this.props.history.push(`/${this.props.lang}/products/new`);

  followUnfollow = async () => {
    try {
      this.setState(() => ({ is_following_load: true }));
      const { is_followed, id, username } = this.state;
      // is_followed = Boolean(is_followed);
      let res = is_followed
        ? await unFollowShop(id)
        : await followShop(id, username);
      const { data = [] } = res;
      if (res)
        this.setState(state => ({
          is_followed: !state.is_followed,
          followers: data,
          n_followers: data.length,
          is_following_load: false
        }));
    } catch (e) {
      this.setState(() => ({ is_following_load: false }));
    }
  };
  openProductDetails = (id, colId) => {
    this.props.TogglePopup(id, this.props.lang_id, colId);
  };
  loadMore = () =>
    this.setState(
      state => ({ page: state.page + 1 }),
      () => this.getProducts(true)
    );

  render() {
    const {
      state: {
        filters = [],
        department,
        products = [],
        filtersBar,
        name,
        avatar,
        cover,
        bio,
        is_mine,
        is_followed,
        no_shop,
        n_followers,
        Color,
        Size,
        Material,
        sort,
        min_price,
        max_price,
        is_busy,
        showMore,
        is_following_load,
        show_loader,
        loading,
        count
      },
      loadMore
    } = this;
    return no_shop ? (
      <Page
        title={this.state.name}
        description={this.state.bio}
        {...this.props}
      >
        <EmptyPage
          locale={this.props.locale}
          {...this.props}
          lang={this.props.lang}
          FouroFour
        />
      </Page>
    ) : (
      <Page
        title={this.state.name}
        description={this.state.bio}
        {...this.props}
      >
        <FiltersMobile
          filter_mobile={this.state.filter_mobile}
          openFilterMobile={this.openFilterMobile}
          closeFilterMobile={this.closeFilterMobile}
          filters={filters}
        />

        <FollowersPopup
          ref={a => (this.FollowersPopup = a)}
          lang={this.props.lang}
          translation={this.props.locale}
          followers={this.state.followers}
        />
        <ActiveModal visible={show_loader}>
          <div className="loader-container">
            <Loader />
          </div>
        </ActiveModal>
        <div className="section">
          <div className="container">
            <Cover
              loading={loading}
              is_followed={is_followed}
              n_followers={n_followers}
              name={name}
              avatar={avatar}
              cover={cover}
              bio={bio}
              isMine={is_mine}
              onClick={this.navigate}
              translation={this.props.locale}
              followUnfollow={this.followUnfollow}
              open_follower_popup={this.openPopUp}
              is_following_load={is_following_load}
              {...this.props}
              open_share={this.openShare}
            />

            <div className="columns">
              <div className="column is-2 is-hidden-mobile">
                <Filters
                  startFilter={this.getProducts}
                  saveToState={this.saveToState}
                  data={filters}
                  department={department}
                  translation={this.props.locale}
                  onClick={this.select_filter}
                  history={this.props.history}
                  withSubCategory={false}
                  {...this.state}
                  {...this.props}
                  is_shop={true}
                />
              </div>

              <div className="column is-10">
                <ProductsGrid
                  count={count}
                  lang={this.props.lang}
                  loading={loading}
                  locale={this.props.locale}
                  is_busy={is_busy}
                  loadMore={loadMore}
                  showMore={showMore}
                  openProductDetails={this.openProductDetails}
                  sort={this.sort}
                  products={products}
                  goToDetails={this.goToDetails}
                  filtersBar={filtersBar}
                  select_filter={this.select_filter}
                  withStatus={is_mine}
                  withTitle
                  match={this.props.match}
                  Search
                />
              </div>
            </div>
          </div>
        </div>
      </Page>
    );
  }
}

/*Shop.propTypes = {
    user: PropTypes.object.isRequired,
    products: PropTypes.array.isRequired,
}*/

export default connect(
  null,
  actions
)(Shop);
