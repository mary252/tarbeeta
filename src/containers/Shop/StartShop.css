.checkbox-text {
  font-family: Rubik;
  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
}

.shop-avatar {
  margin-bottom: 7px;
  width: 140px;
  height: 140px;
  background: #fff;
  border-radius: 50%;
  background-size: contain !important;
  background-position: center !important;
  border: 1px solid #d8d8d8;
  background-repeat: no-repeat !important;
}

.settings-avatar {
  position: relative;
}

.settings-avatar button {
  right: 40px !important;
  top: 5px !important;
}

.shop-cover {
  margin-bottom: 22px;
  height: 300px;
  background-size: cover !important;
  background-position: center !important;
  background-repeat: no-repeat;
  border-radius: 5px;
  border: solid 1px #d8d8d8;
  max-width: 633px;
  background-repeat: no-repeat !important;
  position: relative;
}

.drop-zone {
  height: 100%;
  cursor: pointer;
  background-size: contain !important;
  position: relative;
}

.drop-zone div,
.drop-zone > div > input {
  outline: none !important;
}

.drop-zone > div {
  height: 300px;
}

.drop-zone-content {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  flex-direction: column;
}

.checkbox-text {
  width: 483px;
  /* height: 17px; */
  font-size: 14px;
  color: #565656;
}


/* .white-box-table {
    background: #fff;
    border-radius: 5px;
    padding: 15px 0;
} */

/* 1087px */

@media screen and (max-width: 1088px) {
  .shop-avatar {
    width: 80px;
    height: 80px;
  }

  .shop-cover,
  .drop-zone > div {
    height: 127px;
  }

  .drop-zone-content p {
    text-align: center;
  }

  .saved-address-text {
    margin-right: 12px;
    color: #444444;
  }

  .delete-btn-mobile {
    text-align: right;
  }

  .edit-btn-mobile {
    text-align: left;
  }

}
