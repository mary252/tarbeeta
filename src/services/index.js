export * from "./DepartmentService";
export * from "./categories";
export * from "./subcategory";
export * from "./auth";
export * from "./CartService";
export * from "./helper";
// export * from "./product";
export * from "./shop";
export * from "./wallet";
export * from "./ProductService";
export * from "./ProductDetailsService";
export * from "./PaymentService";
export * from "./AssestService";
export * from "./discover";
export * from "./OrdersService";
export * from "./UserService";
export * from "./GistService";
export * from "./accountsettings";
export * from "./socketIOService";
export * from "./NotificationServices";
