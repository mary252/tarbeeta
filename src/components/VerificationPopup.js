import React, { Component } from "react";
import { Modal } from "../components/Layout";
import { TextInput, TButton } from "./Form";
import {
  sendVerificationEmail,
  sendMobileConfirmationCode
} from "../services/auth";
import { verifyEmail, verifyMobile } from "../services";

class VerificationPopup extends Component {
  state = {
    is_busy: false,
    email_verification_code: null,
    is_sending: false,
    error_msg: null
  };

  toggle = () => {
    this.modal.toggle();
    if (!this.modal.state.visible) {
      this.send_message();
    }
  };

  store_data = (event, input) => {
    this.setState({
      [input]: event.target.value
    });
  };

  verify = async () => {
    this.setState({
      is_busy: true
    });
    const { email_verification_code } = this.state;
    const { isVerifyMobile } = this.props;

    let func = !isVerifyMobile
      ? verifyMobile({ code: email_verification_code })
      : verifyEmail({ code: email_verification_code });
    await func
      .then(response => {
        if (response.status) {
          alert(
            this.props.translation.account_settings_updated_message
              ? this.props.translation.account_settings_updated_message
                  .success_message
              : ""
          );
          window.location.reload();
        }
      })
      .catch(err => {
        this.setState({
          error_msg: err.data ? err.data.message : "",
          is_busy: false
        });
      });
  };

  send_message = async () => {
    this.setState({
      is_sending: true
    });
    try {
      let response =
        this.props.code == 1
          ? await sendMobileConfirmationCode()
          : await sendVerificationEmail();
      if (response.status) {
        this.setState({
          is_sending: false
        });
      }
    } catch (err) {
      this.setState({
        error_msg:
          undefined !== err.response.data.error
            ? err.response.data.error
            : err.response.data.message,
        is_sending: false
      });
    }
  };

  render() {
    const { is_busy } = this.state;
    const { isVerifyMobile, translation } = this.props;
    return (
      <Modal ref={m => (this.modal = m)}>
        <div onClick={e => e.stopPropagation()} className="address-popup">
          <div className="address-popup-div">
            <p className="address-popup-header">
              {translation.verify_mobile.title}
            </p>
            <div className="columns is-mobile">
              <TextInput
                type="number"
                onKeyUp={e => this.store_data(e, "email_verification_code")}
                onKeyPress={e => {
                  if (e.key === "Enter") {
                    this.verify();
                  }
                }}
                placeholder={
                  translation.email_verification.popup_input_placeholder
                }
              />
            </div>

            <div className="columns is-mobile">
              <div className="column is-4">
                <TButton
                  text={translation.cancel}
                  onPress={this.toggle}
                  className="red outline full"
                  ariaLabel="Cancel"
                />
              </div>
              <div className="column is-2 " />
              <div className="column is-6">
                <TButton
                  text={translation.email_verification.verify_button}
                  onPress={() => this.verify()}
                  className="grad-blue full button"
                  loading={is_busy}
                  ariaLabel="Verify"
                />
              </div>
            </div>
          </div>
        </div>
      </Modal>
    );
  }
}

export default VerificationPopup;
