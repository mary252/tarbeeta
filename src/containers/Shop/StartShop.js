import React, { Component } from "react";
import "./StartShop.css";
import { isLoggedIn } from "../../ultils/auth";
import { uploadPic, postNewShop } from "../../services";
import { Redirect } from "react-router-dom";
import { connect } from "react-redux";
import { Page, ShopForm } from "../../components";
import { validate } from "../../ultils/validate/validation";

import * as actions from "../../actions";

const mapStateToProps = state => {
  const {
    user: { mobile_verified, verified, mobile, shop_id, shop_username }
  } = state;
  return {
    mobile_verified,
    verified,
    mobile,
    shop_id,
    shop_username
  };
};

export class StartShop extends Component {
  state = {
    loggedin: false,
    cover: "",
    avatar: "",
    name: "",
    bio: "",
    username: "",
    avatarPreview: "",
    coverPreview: "",
    mobile: "",
    mobileError: "",
    is_busy: false,
    isError: false,
    redirect: false,
    agreeTerms: true,
    nameError: "",
    usernameError: "",
    verify_robot: false
  };

  verify_robot = value => {
    this.setState({
      verify_robot: true
    });
  };

  toggle_busy = () => this.setState({ is_busy: !this.state.is_busy });
  componentWillMount() {
    if (isLoggedIn()) {
      this.setState({
        loggedin: true
      });
    }
  }

  onDrop = e => {
    const fileList = e.dataTransfer ? e.dataTransfer.files : e.target.files;

    this.setState({
      cover: fileList[0]
    });
    uploadPic(e, cover => {
      this.setState({
        coverPreview: cover
      });
    });
  };

  changeAvatar = e => {
    this.setState({
      avatar: e.target.files[0]
    });
    uploadPic(e, avatar => {
      this.setState({
        avatarPreview: avatar
      });
    });
  };

  removeCover = () => this.setState({ cover: null, coverPreview: null });

  validation = fieldsArray => {
    this.setState({
      error_msg: null
    });

    return new Promise(async (resolve, reject) => {
      const errorsArray = await fieldsArray.map((field, i) => {
        return { [`${field}Error`]: validate(field, this.state[field]) };
      });

      let hasErrors = false;

      errorsArray.map((error, j) => {
        let key = Object.keys(error)[0],
          value = Object.values(error)[0];

        this.setState({
          [key]: value
        });

        if (value !== null) {
          hasErrors = true;
        }
      });

      if (hasErrors) {
        reject(false);
      }

      resolve(true);
    });
  };

  startYourShop = async () => {
    // if(this.state.verify_robot){
    this.validation(["name", "mobile"])
      .then(async () => {
        if (this.props.mobile == null) {
          localStorage.setItem("destination", "start");
          this.setState({
            redirect: true
          });
        } else {
          if (!this.props.mobile_verified) {
            this.openPopUp();
          } else {
            const { cover, avatar, name, bio, username, mobile } = this.state;
            const { locale } = this.props;
            this.toggle_busy();
            var data = new FormData();
            data.append("name", name);
            data.append("shop_phone", mobile);
            // this.state.username !== '' ? data.append("username", username) : undefined
            // this.state.bio !== '' ? data.append("bio", bio) : undefined
            // this.state.avatar !== '' ? data.append("avatar", avatar) : undefined
            // this.state.cover !== '' ? data.append("cover", cover) : undefined

            if (this.state.username !== "") {
              data.append("username", username);
            }
            if (this.state.bio !== "") {
              data.append("bio", bio);
            }
            if (this.state.avatar !== "") {
              data.append("avatar", avatar);
            }
            if (this.state.cover !== "") {
              data.append("cover", cover);
            }

            let lang_id = this.props.lang_id;
            // const values = await postNewShop(data, lang_id);

            try {
              let values = await postNewShop(data, lang_id);

              this.setState({
                message: locale.shop_containers.status_200,
                isError: false
              });
              const {
                data: { shop_id = "", shop_username = "" } = {}
              } = values;
              this.props.set_shop_id(shop_id);
              this.toggle_busy();

              if (username == "") {
                setTimeout(() => {
                  this.props.history.push(
                    `/${this.props.lang}/shop/${shop_username}`
                  );
                }, 3000);
              } else {
                setTimeout(() => {
                  this.props.history.push(
                    `/${this.props.lang}/shop/${username}`
                  );
                }, 3000);
              }
            } catch (e) {
              console.log(e);
              this.setState({ isError: true });
              if (e.status == 400) {
                this.setState({
                  message: locale.shop_containers.status_400
                });
              } else if (e.status == 500) {
                this.setState({
                  message: locale.shop_containers.status_500
                });
              } else if (e.status == 422) {
                this.setState({
                  message: locale.shop_containers.status_422
                });
              }
              this.toggle_busy();
            }
          }
        }
      })
      .catch(formErrors => {});
  };

  onChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  VerificationPopup = null;
  openPopUp = () => {
    this.VerificationPopup.toggle();
  };

  removeCover = () => this.setState({ cover: "", coverPreview: "" });

  render() {
    if (!this.state.loggedin) {
      let path = this.props.location.pathname;
      let redirect = path.substring("4");
      return <Redirect to={`/${this.props.lang}/login?src=${redirect}`} />;
    }
    if (this.state.redirect) {
      return <Redirect to={`/${this.props.lang}/mobile`} />;
    }
    const {
      state: {
        cover,
        avatar,
        avatarPreview,
        coverPreview,
        is_busy,
        message,
        isError,
        agreeTerms,
        mobile,
        mobileError
      },
      props: { mobile_verified, verified, locale }
    } = this;

    if (this.props.shop_id) {
      return <Redirect to={`/${this.props.lang}/shop/edit`} />;
    }

    return (
      <Page
        title={locale.shop_containers.start_shop_page_title}
        description={locale.shop_containers.start_shop_page_desc}
        {...this.props}
      >
        {/* <VerificationPopup
          ref={a => (this.VerificationPopup = a)}
          lang={this.props.lang}
          translation={this.props.locale}
          code={1}
        /> */}
        <ShopForm
          {...this.props}
          removeCover={this.removeCover}
          startYourShop={this.startYourShop}
          username={this.state.username}
          onChange={this.onChange}
          validation={this.validation}
          nameError={this.state.nameError}
          usernameError={this.state.usernameError}
          onDrop={this.onDrop}
          changeAvatar={this.changeAvatar}
          avatar={this.state.avatar}
          avatarPreview={this.state.avatarPreview}
          cover={this.state.cover}
          coverPreview={this.state.coverPreview}
          is_busy={this.state.is_busy}
          message={this.state.message}
          isError={this.state.isError}
          locale={this.props.locale}
          onClose={() => this.setState({ message: null })}
          mobile={mobile}
          mobileError={mobileError}
          startShop
        />
      </Page>
    );
  }
}

export default connect(
  mapStateToProps,
  actions
)(StartShop);
