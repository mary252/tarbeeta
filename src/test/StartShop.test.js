import React from "react";
import { shallow } from "enzyme";
import { StartShop } from "../containers/Shop/StartShop";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { baseURL } from "../common/Config";
import { postNewShop } from "../services";
import axios from "axios"; // v0.15.3
import httpAdapter from "axios/lib/adapters/http";
import nock from "nock";
import VerificationPopup from "../components/VerificationPopup";
configure({ adapter: new Adapter() });
jest.setTimeout(30000);

const match = {
  params: {
    lang: "en" //any id you want to set
  }
};
const location = {
  search: "",
  pathname: "/en/start"
};
const host = baseURL;

axios.defaults.host = host;
axios.defaults.adapter = httpAdapter;
let query = {
  lang_id: 1
};
global.URL.createObjectURL = jest.fn();
const Shop = shallow(
  <StartShop
    match={match}
    mobile_verified={1}
    mobile={"0102357890"}
    location={location}
  />
);
let instance = Shop.instance();

describe('start shop render',()=>{
    it('reneders',()=>{
        expect(Shop.exists()).toBeTruthy()
    })
    it("renders correctly", () => {
        expect(Shop).toMatchSnapshot();
      });
    it('the locale load',()=>{
        expect(instance.state.locale).not.toBeNull()
    })

   
       
    it('verify robot function test ',()=>{
        expect(instance.state. verify_robot).toBeFalsy()
        instance.verify_robot('12321')
        expect(instance.state. verify_robot).toBeTruthy()

    })

    it('toogle busy test ',()=>{
        expect(instance.state.is_busy).toBeFalsy()
        instance.toggle_busy()
        expect(instance.state.is_busy).toBeTruthy()
    })
    it('change avatar function test',()=>{
        expect(instance.state.avatar).toBe("")
        let e={
            target:{
                files:['file']
            }
        }
        instance.changeAvatar(e)
        expect(instance.state.avatar).toBe("file")
    })

    it('ondrop function test',()=>{
        expect(instance.state.cover).toBe('')
        let e={
            target:{
                files:['file']
            }
        }
        instance.onDrop(e)
        expect(instance.state.cover).toBe('file')
    })

    it('onchange function test',()=>{
        expect(instance.state.name).toBe('')
        let e={
            target:{
                name:'name',
                value:'name'
            }
        }
        instance.onChange(e)
        expect(instance.state.name).toBe('name')
    })

    
})

describe('start shop functional testing',()=>{
  
    it('name validate',async()=>{
        instance.setState({
            name:'name'
        })
       let res=await instance.validation(['name'])
       expect(res).toBe(true)
       
    })
    it('redirect to mobile page if there is no mobile number found',async ()=>{
        const startShop_copy = shallow(<StartShop match={match} location={location}/>);
        let another_instance=startShop_copy.instance()
        another_instance.setState({
            name:'name',
            verify_robot:true
        })
        await another_instance.startYourShop()
        expect(another_instance.state.redirect).toBeTruthy()
        let dis=localStorage.getItem("destination")
        expect(dis).toEqual('start')
    })
    it('open popup if there is mobile number but not verified',async()=>{
        const startShop_copy = shallow(<StartShop match={match} mobile={'0102357890'} location={location}/>);
        let another_instance=startShop_copy.instance()
        const verification= shallow( <VerificationPopup
            lang={match.params.lang}
            translation={another_instance.state.locale}
            code={1}
          />)
          another_instance.VerificationPopup=verification.instance()
        another_instance.setState({
            name:'name',
            verify_robot:true
        })
        const spyPreventDefault = jest.spyOn(another_instance, 'openPopUp');
        await another_instance.startYourShop()
        expect(spyPreventDefault).toHaveBeenCalled()

    })
    it('start shop work successfully', async()=>{
        instance.setState({
            cover:'cover', 
            avatar:'avatar',
            name:"shop",
            bio:'bio',
            username:'username',
        })
        nock(host)
        .post("/shop")
        .query(query)
        .reply(200, {
          data: {
            "access_token":'3242576786543567897654345678',
            avatar:'awaver',
            user_id:233,
            verified:1, 
            mobile_verified:1
          }
        });
        await instance.startYourShop()
       // expect(postNewShop).toBeCalled()

        expect(instance.state.is_Error).toBeFalsy()
    })
})

describe("buttons test", () => {
  it("change avatar test", () => {
    instance.setState({ loggedin: true });
    const avatar_button = Shop.find("#avatar-button");
    expect(avatar_button.length).toEqual(1);
    // const spy = jest.spyOn(instance, "changeAvatar");
    // avatar_button.prop("onPress")();
    // expect(spy).toHaveBeenCalled()
  });

  // it('name input test',()=>{
  //     const username=Shop.find("#name-input")
  //     expect(username.length).toEqual(1);
  //     const event = { target: { name: "name", value: "456" } };
  //     username.simulate("keyUp", event);
  //     expect(instance.state.email).toEqual("456");

  //   })
});
