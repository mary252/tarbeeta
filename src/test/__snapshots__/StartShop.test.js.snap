// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`start shop render renders correctly 1`] = `ShallowWrapper {}`;
